### DISCUSSION TITLE
Query syntax for consecutive sibling nodes?

### DISCUSSION THREAD

**Discussion by euclidianAce** [tree-sitter/tree-sitter#750] [Q&A] [answered]
The documentation seems to be incorrect or I am misinterpreting it.

```
Grouping Sibling Nodes

You can also use parentheses for grouping a sequence of sibling nodes. For example, this pattern would match a comment followed by a function declaration:

(
  (comment)
  (function_declaration)
)
```

I would expect this query to match against comments followed by function declarations, but when run in the playground with some simple C code
```c

// this is a comment, not followed immediately by a function, but eventually followed by a function
static const int x = 2;

// this is a comment, followed by a function
int main(void) {
    return 0;
}

// this is just a comment, at the end of the code
```

when running the query 
```
(
  (comment) @a
  (function_definition) @b
) @c
```
in the playground, the two top comments are matched, despite the first comment and `main` having a node between them. This makes sense, I guess, but is there query syntax for named nodes directly next to each other?

**Comment by vigoux** [comment, marked as answer]
Hi, I think that what you are searching for is the `.` syntax, and that the query you want is : 
```
(
  (comment) @a
  .
  (function_definition) @b
) @c
```

**Comment by euclidianAce** [comment]
Thanks! Is there somewhere where this is documented? Or is the query language not fully documented yet?

**Comment by stsewd** [comment]
Yeah, there are some undocumented features https://github.com/tree-sitter/tree-sitter/pull/549
https://github.com/tree-sitter/tree-sitter/pull/738

I opened a PR documented some of them https://github.com/tree-sitter/tree-sitter/pull/713

**Comment by jmehnle** [comment]
How do you make `(comment) @a` **optional** in the above example so the `(function_definition)` can match with or without a preceding comment?

```
(
  (comment)? @a
  .
  (function_definition) @b
) @c
```

… doesn't work, because when Tree Sitter attempts a match with `(comment) @a` **absent**, the `.` adopts the meaning of "immediately following whatever pattern precedes `@c` in the query, or the top of the tree.
